apps/dashboard/app/models/project.rb

```diff
@@ -27,33 +28,69 @@ def find(project_path)
     def dataroot
       Rails.logger.debug("project path is: #{OodAppkit.dataroot.join('projects')}")
 
-      OodAppkit.dataroot.join('projects').tap do |p|
-        p.mkpath unless p.exist?
+      OodAppkit.dataroot.join('projects').tap do |path|
+        path.mkpath unless path.exist?
       rescue StandardError => e
         Pathname.new('')
       end
     end
   end
 
+  validates :dir, presence: true
+  validates :dir, format: {
+
+    with: /[\w-]+\z/,
```

Comment:
This still needs adjust so it doesn't allow things like `kl;akje_kljdkj\\kljekj`